syntax = "proto3";

option go_package = "./cmds";

service Agent {
  rpc RunEchoCommand (stream EchoCommandResponse) returns (stream EchoCommandRequest) {}
  rpc RunShellCommand (stream ShellCommandResponse) returns (stream ShellCommandRequest) {}
}

service AgentManager {
  rpc RunEchoCommand (EchoCommandRequest) returns (EchoCommandResponse) {}
  rpc RunShellCommand (ShellCommandRequest) returns (ShellCommandResponse) {}
  rpc GetAgents(Empty) returns (stream AgentInfo) {}
}

message ShellCommandRequest {
  string Addr = 1;
  string cmd = 2;
}

message ShellCommandResponse {
  bytes output = 1;
}

message AgentInfo {
  string Addr = 1;
}

message EchoCommandRequest {
  string Addr = 1;
  string text = 2;
}

message EchoCommandResponse {
  string Addr = 1;
  string text = 2;
}

message Empty {}
